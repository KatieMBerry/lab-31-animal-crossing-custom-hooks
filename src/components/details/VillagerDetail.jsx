import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

const VillagerDetail = ({ image, name, quote, skill }) => (
    <figure>
        <img src={image} alt={name} />
        <figcaption>
            <p>{name}</p>
            <p>Quote: {quote}</p>
            <p>Skill: {skill}</p>
        </figcaption>
    </figure>
);

VillagerDetail.propTypes = {
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    quote: PropTypes.string.isRequired,
    skill: PropTypes.string.isRequired
}

export default VillagerDetail;
