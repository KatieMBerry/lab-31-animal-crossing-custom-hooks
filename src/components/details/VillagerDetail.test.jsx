import React from 'react';
import { render, cleanup } from '@testing-library/react';
import VillagerDetail from './VillagerDetail';

describe('VillagerDetail component', () => {
    afterEach(() => cleanup());
    it('renders VillagerDetail', () => {
        const { asFragment } = render(<VillagerDetail />);
        expect(asFragment()).toMatchSnapshot();
    });
});
