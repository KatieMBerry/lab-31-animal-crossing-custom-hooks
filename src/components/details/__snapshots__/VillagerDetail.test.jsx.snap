// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VillagerDetail component renders VillagerDetail 1`] = `
<DocumentFragment>
  <figure>
    <img />
    <figcaption>
      <p />
      <p>
        Quote: 
      </p>
      <p>
        Skill: 
      </p>
    </figcaption>
  </figure>
</DocumentFragment>
`;
